src/Microsoft.DotNet.Tools.Compiler.Native/IntermediateCompilation/Mac/MacRyuJitCompileStep.cs

```diff
@@ -0,0 +1,146 @@
+
+        public MacRyuJitCompileStep(NativeCompileSettings config)
+        {
+            this.config = config;
+            InitializeArgs(config);
+        }
+
+        public int Invoke()
+        {
+            Console.WriteLine("MacRyuJitCompileStep: Invoking JIt step");
+            var result = InvokeCompiler();
+            if (result != 0)
+            {
+                Reporter.Error.WriteLine("Compilation of intermediate files failed.");
+            }
+
+            return result;
+        }
+
+        public bool CheckPreReqs()
+        {
+            // TODO check for clang
+            return true;
+        }
+
+        private void InitializeArgs(NativeCompileSettings config)
+        {
+            var argsList = new List<string>();
+
+            // Flags
+            argsList.Add(cflags);
+            
+            // Add Stubs
+            argsList.Add("-I "+Path.Combine(config.AppDepSDKPath, "CPPSdk/osx.10.10"));
+            argsList.Add("-I "+Path.Combine(config.AppDepSDKPath, "CPPSdk"));
+            argsList.Add(Path.Combine(config.AppDepSDKPath, "CPPSdk/osx.10.10/osxstubs.cpp"));
+
+            // Input File
+            var inLibFile = DetermineInFile(config);
+            argsList.Add("-Xlinker "+inLibFile);
+
+            // Libs
+            foreach (var lib in libs)
+            {
+                var libPath = Path.Combine(config.IlcPath, lib);
+                argsList.Add("-Xlinker "+libPath);
+            }
+
+            // AppDep Libs
+            var baseAppDepLibPath = Path.Combine(config.AppDepSDKPath, "CPPSdk/osx.10.10", config.Architecture.ToString());
+            foreach (var lib in appdeplibs)
+            {
+                var appDepLibPath = Path.Combine(baseAppDepLibPath, lib);
+                argsList.Add("-Xlinker "+appDepLibPath);
+            }
+
+            // Output
+            var libOut = DetermineOutputFile(config);
+            argsList.Add($"-o \"{libOut}\"");
+
+            this.CompilerArgStr = string.Join(" ", argsList);
+        }
+
+        private int InvokeCompiler()
+        {
+            var result = Command.Create(CompilerName, CompilerArgStr)
+                .ForwardStdErr()
+                .ForwardStdOut()
+                .Execute();
+
+            // Needs System.Native.dylib PAL in output - ideally, this should come as app-dependency,
+            // after the app is native compiled, by dotnet-compile.
+            var sharedLibPath = Path.Combine(config.IlcPath, "System.Native.dylib");
+            var outputSharedLibPath = Path.Combine(config.OutputDirectory, "System.Native.dylib");
+            try
+            {
+                File.Copy(sharedLibPath, outputSharedLibPath);
+            }
+            catch(Exception e)
+            {
+                Reporter.Error.WriteLine("Unable to copy System.Native.dylib to output");
+            }
+
+            return result.ExitCode;
+        }
+
+        private string DetermineInFile(NativeCompileSettings config)
```

Comment:
@brthor, is it okay if it is copy paste for now, so Gaurav can enable our CI to run JIT mode tests? This would help our other devs on the team as we uncovered some really interesting scenarios for the toolchain to address, while enabling Mac JIT mode tests. Bottom line, can you refactor these as part of your ensuing PR?